### PR TITLE
Avoid squeeze for seq_len equal to one.

### PR DESCRIPTION
**Pull request by andreyanufr**
### Changes

Changed squeeze logic for statistics postprocessing. Some models has internal processing which leads to variational seq_len and in this case we have statistics with different dimension if one off them has seq_len equal to one.

### Reason for changes

Bug with AWQ/SE for mixtral-8x7b-v0.1

### Tests

tests/openvino/native/quantization/test_weights_compression.py::test_one_dimentional_samples

### CODE REVIEW

File: nncf/quantization/algorithms/weight_compression/algorithm.py [hunk 1]
```diff
@@ -482,6 +482,22 @@ def input_filter_func(point):
         self._fp_inputs[input_id] = input_fp
         return self._fp_inputs[input_id]
 
+    @staticmethod
+    def _get_dynamic_shape(x: List[Tensor]):
```
Comment (ljaljushkin) [inline nncf/quantization/algorithms/weight_compression/algorithm.py:486]: Please provide a context in the docstring, why this function is needed.

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: nncf/quantization/algorithms/weight_compression/algorithm.py (hunk 1 above)
Comment (andreyanufr) [reply nncf/quantization/algorithms/weight_compression/algorithm.py:486]: Done

**Review by ljaljushkin — APPROVED** [review]

**Review by alexsu52 — COMMENTED** [review]
LGTM, please provide weight compression test results.

**Comment by KodiaqQ** [comment]
When this PR will be merged? Is it still blocked, @alexsu52?

**Comment by alexsu52** [comment]
post_training_weight_compression/136/

**Comment by andreyanufr** [comment]
> @andreyanufr, please, investigate degradation in post_training_weight_compression/136/

@alexsu52 post_training_weight_compression/156

**Review by KodiaqQ — APPROVED** [review]

**Pull request merged**